Metadata refactor

muXxer opened the pull request:
- All Metadata features are now `map[string][]byte` instead of `[]byte`
- Removed `StateMetadata` field in favor of new `StateMetadataFeature`

depends on https://github.com/iotaledger/hive.go/pull/600

alexsporn commented on gen/feat_metadata.tmpl:
```diff
@@ -0,0 +1,83 @@
+	"github.com/iotaledger/hive.go/serializer/v2"
+)
+
+type {{.Name}}Entries map[string][]byte
```
There is no mention whatsoever about ordering of the map. This makes serialization non-deterministic.

alexsporn replied:
This is now also forcing the use of a key everytime. (e.g. `data` as in every changed testcase)

alexsporn replied:
Why is this better than something like IRC-27 or IRC-30? The protocol does not need the map here to function.

muXxer replied:
The new serialization of maps is deterministic in serix. So this wouldn't be a problem anymore :+1:

muXxer committed:
Metadata refactor
- All Metadata features are now `map[string][]byte` instead of `[]byte`
- Removed StateMetadata in favor of new GovernorMetadata feature

muXxer committed:
Implement the changes discussed in the call

muXxer committed:
Add checks for non ASCII chars in metadata keys

muXxer committed:
Update hive.go for correct UTF8 validity checks in serix

muXxer committed:
Add test for empty keys in metadata

muXxer committed:
Fix metadata description
